cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/ManagementClusterInventory.tsx

```diff
@@ -13,67 +13,84 @@ import { ManagementCluster } from '../../swagger-api/models/ManagementCluster';
 import { ManagementService } from '../../swagger-api';
 import { NavRoutes } from '../../shared/constants/NavRoutes.constants';
 import './ManagementClusterInventory.scss';
+import PageNotification, { Notification, NotificationStatus } from '../../shared/components/PageNotification/PageNotification';
 
 function ManagementClusterInventory() {
+    const [showLoading, setShowLoading] = useState<boolean>(false);
+    const [notification, setNotification] = useState<Notification | null>(null);
     const [managementClusters, setManagementClusters] = useState<ManagementCluster[]>([]);
     const [showDeleteModal, setShowDeleteModal] = useState<boolean>(false);
     const [clusterNameForDeletion, setClusterNameForDeletion] = useState<string>('');
 
     const navigate = useNavigate();
 
-    const retrieveManagementClusters = function () {
-        ManagementService.getMgmtClusters().then(
-            (data) => {
-                setManagementClusters(data);
-            },
-            (err) => {
-                console.warn(`management clusters get api failed with error: ${err}`);
-            }
-        );
-    };
+    // sets notification to null to dismiss alert
+    function dismissAlert() {
+        setNotification(null);
+    }
 
-    // Retrieve management clusters list on page load
+    // Retrieve management clusters list on page load; then set loading to false
     useEffect(() => {
-        retrieveManagementClusters();
+        retrieveManagementClusters().then(() => {
+            console.info('initial loading of management clusters list completed');
+        });
     }, []);
 
+    const retrieveManagementClusters = async () => {
+        setShowLoading(true);
+        try {
+            const data = await ManagementService.getMgmtClusters();
+            setManagementClusters(data);
+        } catch (e) {
+            setNotification({
+                status: NotificationStatus.DANGER,
+                message: `Unable to retrieve Management Clusters: ${e}`,
+            } as Notification);
+        } finally {
+            setShowLoading(false);
+        }
+    };
+
     // Helper function returns true if management clusters exist; otherwise returns false
-    const hasManagementClusters = function () {
+    function hasManagementClusters() {
```

Comment:
nit about the next line (which is not part of your PR): can we just do `return managementCluster.length > 0`?

Reply:
yep, will update that thanks